topics/plugin-preprocess-xslt.dita

```diff
@@ -0,0 +1,72 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE task PUBLIC "-//OASIS//DTD DITA Task//EN" "task.dtd">
+<!--  This file is part of the DITA Open Toolkit project. See the accompanying LICENSE file for applicable license.  -->
+<task id="plugin-xsltparams">
+  <title>Processing topics with XSLT in preprocess</title>
+  <shortdesc>You can pass parameters from the Ant build to existing XSLT steps in both the pre-processing pipeline and
+    certain DITA-OT transformations. This can be useful if you want to make the parameters available as global
+      <xmlelement>xsl:param</xmlelement> values within XSLT overrides.</shortdesc>
+  <taskbody>
+    <steps>
+      <step>
+        <cmd>Create a plug-in descriptor file <filepath>plugin.xml</filepath> that imports a new Ant buildfile
```

Comment:
Do we have a topic somewhere that describes how to write a plug-in? Basically there are two ways:

1. Create a directory somewhere and that's your plug-in source directory/repo.
2. Create a plug-in directory under `$DITA_HOME/plugins`.

The latter is a quick and dirty but simple way of doing it. Then there's the hacker version

3. Create a directory somewhere and that's your plug-in source directory/repo and then use `ln` command to create a symbolic link to plug-in directory under `$DITA_HOME/plugins`.

Reply:
> Do we have a topic somewhere that describes how to write a plug-in?

In [Plug-in coding conventions](https://www.dita-ot.org/dev/topics/plugin-coding-conventions.html), we suggest a variant of № 1 above:

> Store the source code of your plug-ins outside of the DITA-OT installation directory, and add the repository location to the list of plug-in directories defined in the `plugindirs` entry of the `configuration.properties` file.

We [suggest this approach](https://www.dita-ot.org/dev/parameters/configuration-properties-file.html#configuration__plugindirs) for testing the sample plug-ins provided with the docs.

I use this configuration properties approach, and symbolic links.